File: packages/go/domains.go
package hubfly

import (
	"context"
	"fmt"
)

type DomainsService struct {
	client *Client
}

func (s *DomainsService) List(ctx context.Context) (*Response[[]Domain], error) {
	return Request[[]Domain](ctx, s.client, "GET", "/api/v1/domains", nil)
}

func (s *DomainsService) Create(ctx context.Context, params CreateDomainParams) (*Response[Domain], error) {
	return Request[Domain](ctx, s.client, "POST", "/api/v1/domains/create", params)
}

func (s *DomainsService) Verify(ctx context.Context, domainID string) (*Response[map[string]bool], error) {
	path := fmt.Sprintf("/api/v1/domains/%s/verify", domainID)
	return Request[map[string]bool](ctx, s.client, "POST", path, nil)
}

func (s *DomainsService) Remove(ctx context.Context, domainID string) (*Response[map[string]bool], error) {
	path := fmt.Sprintf("/api/v1/domains/%s/remove", domainID)
	return Request[map[string]bool](ctx, s.client, "POST", path, nil)
}

File: packages/ts/src/modules/domains.ts
import type { HubflyClient } from '../client.js';
import type { ApiResponse, Domain, CreateDomainParams } from '../types.js';

export class DomainsModule {
  constructor(private readonly client: HubflyClient) {}

  /**
   * List custom domains
   */
  public async list(): Promise<ApiResponse<Domain[]>> {
    return this.client.request<Domain[]>('/api/v1/domains');
  }

  /**
   * Register a custom domain
   */
  public async create(params: CreateDomainParams): Promise<ApiResponse<Domain>> {
    return this.client.request<Domain>('/api/v1/domains/create', {
      method: 'POST',
      body: params,
    });
  }

  /**
   * Trigger DNS verification check for a domain
   */
  public async verify(domainId: string): Promise<ApiResponse<{ verified: boolean }>> {
    return this.client.request<{ verified: boolean }>(`/api/v1/domains/${domainId}/verify`, {
      method: 'POST',
    });
  }

  /**
   * Remove a custom domain
   */
  public async remove(domainId: string): Promise<ApiResponse<{ removed: boolean }>> {
    return this.client.request<{ removed: boolean }>(`/api/v1/domains/${domainId}/remove`, {
      method: 'POST',
    });
  }
}
